yt_dlp/extractor/common.py

```diff
@@ -2893,11 +2933,11 @@ def add_segment_url():
                     if content_type in ('video', 'audio', 'image/jpeg'):
                         f['manifest_stream_number'] = stream_numbers[f['url']]
                         stream_numbers[f['url']] += 1
-                        formats.append(f)
+                        period_entry['formats'].append(f)
                     elif content_type == 'text':
-                        subtitles.setdefault(lang or 'und', []).append(f)
-
-        return formats, subtitles
+                        period_entry['subtitles'][lang or 'und'].append(f)
+            if period_entry['formats'] or period_entry['subtitles']:
+                yield period_entry
```

Comment:
Is the if condition necessary? This should work just as well
```suggestion
            yield period_entry
```

Reply:
No, indeed, since the formats and subtitles are initialized at the beginning. Removed.